src/pipeline.rs

```diff
@@ -220,6 +220,25 @@ where
         self.pump(crate::pumps::enumerate::EnumeratePump)
     }
 
+    /// Attach a batch pump to the pipeline. Batch will collect `n` items in the pipeline and emit a `Vec`.
```

Comment:
please add a clarification that while the buffer is not full, no items will be emitted, and another clarification that if the pipeline input is finite,  the last item emitted will be partial